app/dashboard/src/pages/authentication/Setup/Setup.tsx

```diff
@@ -85,16 +86,34 @@ const BASE_STEPS: Step[] = [
             })
           }
           defaultValues={{ username: defaultName }}
-          onSubmit={({ username }) => {
-            // If user is already created we shouldn't call `setUsername` if value wasn't changed
-            if (username === defaultName && isUserCreated) {
-              goToNextStep()
-              return
-            } else {
-              return setUsername(username).then(() => {
-                goToNextStep()
-              })
+          onSubmit={async ({ username }) => {
+            // If user is already created we shouldn't call `setUsername` if the value has not been
```

Comment:
Why do we check for subscription on `SetUsername` step?

Reply:
we don't, we check for registration. but good catch on the error message

Reply:
![CleanShot 2024-09-25 at 12 25 42@2x](https://github.com/user-attachments/assets/9f0c864e-48fe-4774-8b9e-829d86150ccc)

The error message is indeed miseading. Also, I'm not sure we need that at all, could you provide more context on why this snippet exists?